### PR TITLE
Fixes #27660 - adds a doc for legacy js in storybook

### PR DESCRIPTION
**Pull request by amirfefer**
This doc explains how to access new js code in old js files, and how to observe the store for changes.

### CODE REVIEW

File: webpack/stories/docs/LegacyJs.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+  else {
+    unsubscribe();
+  }
+}
+
+```
+This function have two paramteres as well:
+1. the observed store.
+2. an unsubscribe functopn to stop the observation.
```
Comment (xprazak2) [inline webpack/stories/docs/LegacyJs.md]: s/functopn/function/

Are there any recommendations regarding `unsubscribe`? Should we always unsubscribe at some point or is it ok to stay subscribed?

**Comment by xprazak2** [comment]
Reads well, I left one comment inline.

File: webpack/stories/docs/LegacyJs.md (hunk 1 above)
Comment (amirfefer) [reply webpack/stories/docs/LegacyJs.md]: it's optional, I'll add it to the doc :)

**Comment by amirfefer** [comment]
Thanks @xprazak2  !
I've updated the doc :)

**Comment by amirfefer** [comment]
# Legacy JS

Foreman's legacy javascript is based on ruby on rails assets pipeline and located in `assets/javascripts`


### Access webpack javascript
In order to access new js logic in old js files, we created a global object -`tfm`, which contains a set of functions and located in `/webpack/assets/javascripts/bundle.js`
Please use this object instead of using the `window` object directly.

### Observing the store

With `observeStore` you can observe for changes of the store:

```js
 tfm.store.observeStore('notifications.items', tfm.doSomething);
 ```
`observeStore` accepts two parameters:
1. the part of the store to be observed.
2. a function to run when a change is detected.

```js
const doSomething = (items, unsubscribe) => {
  if (items.length) {
    doSomething();
  }
  else {
    unsubscribe();
  }
}
```
This function have two paramteres as well:
1. the observed store.
2. an unsubscribe function to stop the observation (optional).

File: webpack/stories/index.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import addingDependencies from './docs/addingDependencies.md';
 import internationalization from './docs/internationalization.md';
 import plugins from './docs/plugins.md';
 import SlotAndFill from './docs/SlotAndFill.md';
+import LegacyJs from './docs/LegacyJS.md';
```
Comment (xprazak2) [inline webpack/stories/index.js]: should be `./docs/LegacyJs.md`, otherwise the file is not found.

Comment (amirfefer) [reply webpack/stories/index.js]: Thanks, I renamed it and forgot to update the import...

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**